### ISSUE TITLE
Handle copying directory into itself

### ISSUE THREAD

**Issue by nabijaczleweli** [thecoshman/http#96]
https://github.com/thecoshman/http/pull/95#discussion_r370556596

Ref: #61 
Ref: #95

**Comment by thecoshman** [comment]
Are we needing to make a call about how to handle this? I think I'm ok with having this simple not a thing you can do, but don't really see much reason why it can't be supported.

**Comment by nabijaczleweli** [comment]
No, I would just need to copy an algorithm from rsync instead of stealing one from elsewhere
